packages/superset-ui-plugin-chart-icicle/src/Icicle.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { PureComponent } from 'react';
+import { IcicleNode } from '../types/IcicleNode';
+import { default as IciclePath } from './IciclePath';
+import { HierarchyRectangularNode } from 'd3-hierarchy';
+
+type Props = {
+  className?: string;
+  width: number;
+  height: number;
+  data: IcicleNode;
+  boxMargin: {
+    x: number;
+    y: number;
+  };
+  color: (name: string) => string;
+  contentRenderer: () => void;
+  isVertical: boolean;
+  rounding: number;
+  transitionDuration: number;
+};
+
+export default class IcicleChart extends PureComponent<Props> {
+  private chartRef = React.createRef<HTMLDivElement>();
```

Comment:
Worth considering now (since this is the skeleton) -- should this be a functional component, particularly now that giving those lightweight state with hooks is easy? My default functional/not-functional determination is "does this element need direct access to the eventual reducer?"

I haven't thought much about whether or not D3 means classes are way better or not. In any case, it's a judgment call; just wanted to point it out.

Reply:
Good point, I haven't given functional components too much thought in relation to the Icicle Chart and D3. I will look into it and see if it's worth going down that route instead.